robot/impl/local_robot_test.go

```diff
@@ -3238,3 +3240,132 @@ func TestImplicitDepsAcrossModules(t *testing.T) {
 	_, err = r.ResourceByName(motor.Named("m2"))
 	test.That(t, err, test.ShouldBeNil)
 }
+
+func TestResourceByNameAcrossRemotes(t *testing.T) {
+	ctx := context.Background()
+	logger := logging.NewTestLogger(t)
+
+	// Setup a robot1 -> robot2 -> robot3 -> robot4 remote chain. Ensure that if
+	// robot4 has an encoder "e", all robots in the chain can retrieve it by
+	// simple name "e" or short name "[remote-prefix]:e". Also ensure that a
+	// motor "m1" on robot1 can depend on "robot2:robot3:robot4:e" and a motor
+	// "m2" on robot2 can depend on "e".
+
+	addr2 := "localhost:8081"
+	addr3 := "localhost:8082"
+	addr4 := "localhost:8083"
```

Comment:
probably need to handle ports like https://github.com/viamrobotics/rdk/commit/6049c6d73f0906623793a1ff61a56be7bc8a8c96

Reply:
Ah good point; I'll guess we'll have to remember to do this from now on. Modified `startWeb` to use `TryReserveRandomPort` in a loop.